duckduckgo/duckduckgo-privacy-extension
# Whitelist UX changes for the main popup

**Reviewer:** @andrey-p @dharb 
**CC:** @alohaas 



## Description:
Following up from #137, second part of the UX changes: main popup.
Sorry for the messy diff, once #137 is merged it should be fine.

TODO:
- [ ] Pre-populate the whitelist input field with last visited site

<img width="302" alt="screen shot 2018-02-22 at 2 06 30 am" src="https://user-images.githubusercontent.com/3652195/36524721-1141ec18-1775-11e8-890f-fec266cea9c7.png">
<img width="301" alt="screen shot 2018-02-22 at 2 05 45 am" src="https://user-images.githubusercontent.com/3652195/36524722-1150b798-1775-11e8-8656-dc64daef2fd5.png">
<img width="303" alt="screen shot 2018-02-22 at 2 05 27 am" src="https://user-images.githubusercontent.com/3652195/36524723-115d4d00-1775-11e8-8ed4-79e63fd4fee2.png">


## Steps to test this PR:
1. Build and reload the extension
2. On special pages, when the popup is disabled, the "manage whitelist" and "report broken sites" piece should look disabled as well. Clicking on either of those links shouldn't have any effect
3. On any other page, "manage whitelist" should open the options page (to test the whitelist adding/removing functionality, see testing steps in #137), white "report broken site" should open a mailto:link with correct URL and browser information
4. Click on the privacy protection toggle to whitelist a site: there should be an "added to whitelist" message appearing briefly, before reloading the page
5. Open the popup and click again on the toggle button to remove the site from whitelist: this time the page should reload immediately
6. Reset stats or remove and re-install the extension to verify the layout with the no-data graphbar

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
